ovisbot/extensions/ctf/ctf.py

```diff
@@ -124,13 +127,20 @@ async def archive(self, ctx, ctf_name):
         ctf = CTF.objects.get({"name": ctf_name})
         category = discord.utils.get(ctx.guild.categories, name=ctf_name)
         ctfrole = discord.utils.get(ctx.guild.roles, name="Team-" + ctf_name)
-
+        
         if ctfrole is not None:
             await ctfrole.delete()
+
         for c in category.channels:
+            challenge = next((ch for ch in ctf.challenges if ch.name == c.name), None)
+            if challenge is not None and challenge.solved_at:
```

Comment:
Btw I think we should backup the pins for unsolved challenges too since sometimes good findings get pinned while looking at a challenge + writeups in the end. What do you think?

```suggestion
            if challenge is not None:
```

Reply:
True, I forgot about the case where we share writeups afterwards and pin them. Good catch. I believe we should do it.

Reply:
Hello, I changed this to include not solved challenges as well, because as you mentioned, there are valuable findings in those challenges as well.